test/performance/odp_ipsecfwd.c

```diff
@@ -681,9 +728,9 @@ static odp_bool_t send(const pktio_t *pktio, uint8_t index, odp_packet_t pkt)
 	return odp_pktout_send(pktio->out_dir_qs[index], &pkt, 1) == 1;
 }
 
-static odp_bool_t enqueue(const pktio_t *pktio, uint8_t index ODP_UNUSED, odp_packet_t pkt)
+static odp_bool_t enqueue(const pktio_t *pktio, uint8_t index, odp_packet_t pkt)
 {
-	return odp_queue_enq(pktio->out_ev_q, odp_packet_to_event(pkt)) == 0;
+	return odp_queue_enq(pktio->out_ev_qs[index], odp_packet_to_event(pkt)) == 0;
```

Comment:
Since index is CPU ID, this causes reordering of forwarded packets in the ordered mode, defeating much of the purpose of ordered scheduling. There should at least be an option (which should also be the default) to select output queue based on some sort of flow hash (something like xoring the last bytes of the source and destination IP address of the packet would probably be good for this test program).

Reply:
Modified in v2. Something like this, perhaps?